src/_P184_Triac.ino

```diff
@@ -0,0 +1,380 @@
+      // see _P026_Sysinfo.ino for more examples.
+      // event->idx        = P184_OUTPUT_TYPE_INDEX;
+      event->sensorType = Sensor_VType::SENSOR_TYPE_DIMMER; //static_cast<Sensor_VType>(PCONFIG(event->idx));
+      success           = true;
+      break;
+    }
+
+    case PLUGIN_SET_DEFAULTS:
+    {
+      // Set a default config here, which will be called when a plugin is assigned to a task.
+      P184_TRIGGER_CONFIG() = 0;
+      P184_DEAD_ZONE_CONFIG() = 0;
+      P184_ZERO_CROSS_PIN() = -1;
+      P184_TRIGGER_PIN() = -1;
+      PCONFIG(P184_OUTPUT_TYPE_INDEX) = static_cast<uint8_t>(Sensor_VType::SENSOR_TYPE_DIMMER);
+      success = true;
+      break;
+    }
+
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      addRowLabel(F("Trigger Pin"));
+      addPinSelect(PinSelectPurpose::Generic_output, "trigger_pin", P184_TRIGGER_PIN());
+
+      addRowLabel(F("Zero Cross Pin"));
+      addPinSelect(PinSelectPurpose::Generic_input, "zero_crossing_pin", P184_ZERO_CROSS_PIN());
+      {
+        const __FlashStringHelper* optionsEdge[] = { F("RISING"), F("FALLING")};
+        const int optionsValsEdge[] = { RISING, FALLING };
+        constexpr int optionsCountEdge = NR_ELEMENTS(optionsValsEdge);
+        const FormSelectorOptions trigEdgeSelector(optionsCountEdge, optionsEdge, optionsValsEdge);
+        trigEdgeSelector.addFormSelector(F("Interrupt mode"), F("trigger_edge"),  P184_TRIGGER_EDGE_CONFIG());
+      }
+      
+      {
+        const __FlashStringHelper* optionsFreq[] = { F("60Hz"), F("50Hz")};
+        const int optionsValsFreq[] = { P184_60HZ_HALF_WAVE_TIME_US_ONE_PERCENT, P184_50HZ_HALF_WAVE_TIME_US_ONE_PERCENT };
+        constexpr int optionsCountFreq = NR_ELEMENTS(optionsValsFreq);
+        const FormSelectorOptions freqSelector(optionsCountFreq, optionsFreq, optionsValsFreq);
+        freqSelector.addFormSelector(F("Grid frequency"), F("freq"),  P184_FREQ_CONFIG());
+      }
+      
+
+      addFormNumericBox(F("Trigger"), F("trigger"), P184_TRIGGER_CONFIG(), 0, 100);
+      addUnit(F("%"));
+
+      addFormNumericBox(F("Trigger Deadzone"), F("trigger_deadzone"), P184_DEAD_ZONE_CONFIG(), 0, 20);
+      addUnit(F("%"));
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+
+      if (P184_data.zero_crossing_pin != -1)
+      {
+        detachInterrupt(P184_data.zero_crossing_pin);
+      }
+
+      P184_TRIGGER_CONFIG() = getFormItemInt(F("trigger"));
+      P184_DEAD_ZONE_CONFIG() = getFormItemInt(F("trigger_deadzone"));
+      P184_ZERO_CROSS_PIN() = getFormItemInt(F("zero_crossing_pin"));
+      P184_TRIGGER_PIN() = getFormItemInt(F("trigger_pin"));
+      P184_TRIGGER_EDGE_CONFIG() = getFormItemInt(F("trigger_edge"));
+      P184_FREQ_CONFIG() = getFormItemInt(F("freq"));
+
+      P184_data.trigger_value = P184_TRIGGER_CONFIG();
+      // Recalculate power value based on the new trigger value from the form
+      for (int i = 0; i <= 100; ++i) {
```

Comment:
Magic number used here, better would be to use:
`constexpr uint8_t power_to_trigger_lut_size = NR_ELEMENTS(power_to_trigger_lut);`
(just below the `power_to_trigger_lut` array, and the `101` size in that array definition can also be removed. The size is determined at compile-time, this way)
and here, use:
`for (uint8_t i = 0; i < power_to_trigger_lut_size; ++i) {`